model/project_ref.go

```diff
@@ -758,6 +758,8 @@ func (p *ProjectRef) SetGithubAppCredentials(ctx context.Context, appID int64, p
 		if ghApp != nil {
 			return githubapp.RemoveGitHubAppAuth(ctx, ghApp)
 		}
+		// No-op if the passed in values are empty and no entry exists.
+		return nil
```

Comment:
I made this no-op instead of error because there isn't an easy way to check and clear if it exists without a lot of additional code.

Reply:
Could we reword the comment a bit? It sounds more like this is an "exit early" (which is valid since it already attempted to remove the app auth), and it seems like the intent of the method is to conveniently delete the app creds if the app ID and key are both unset.